### PR TITLE
Clean search-domain API for faster loads

### PR DESCRIPTION
**Pull request by valentinludu**
Clean `search-domain` and `search-name` API for faster loads.

Conclusions:
No token gates will be returned from space endpoints.
Places where we need token gates, will be loaded from the `token-gates` endpoint

### CODE REVIEW

**Review by msadura — APPROVED** [review]

**Review by mattcasey — CHANGES_REQUESTED** [review]
can the frontend be changed so that we dont mutate the state of another api request? this can be overridden

**Comment by valentinludu** [comment]
Done!

> can the frontend be changed so that we dont mutate the state of another api request? this is not foolproof and can be overridden

**Review by mattcasey — CHANGES_REQUESTED** [review]
We still are kind of combining space and token gates into one object with i think will result in bugs. I would question if we even nee to return tokenGates in the spaces context anymore, and could get rid of the SpaceWithGates interface. This would highlight some other areas we need to address including these components that use `SpaceAccessGate: SpaceAccessGateWithSearch and RewardSignupButton. It seems like SpaceAccessGate ought to be able to just query token gates on its own?

File: charmClient/hooks/tokenGates.ts [hunk 1]
```diff
@@ -9,8 +9,8 @@ import type { TokenGate, TokenGateWithRoles, TokenGateConditions } from 'lib/tok
 
 import { useDELETE, useGET, usePOST, usePUT } from './helpers';
 
-export function useGetTokenGates(spaceId: string) {
-  return useGET<TokenGateWithRoles[]>(spaceId ? '/api/token-gates' : null, { spaceId });
+export function useGetTokenGates(spaceId: string, spaceDomain?: string) {
```
Comment (mattcasey) [inline charmClient/hooks/tokenGates.ts:12]: i dont see a need to also query by spacedomain, please remove this and only use spaceId

**Review by mattcasey — CHANGES_REQUESTED** [review]
I think there are other places where we get token gates, also please see if you can kill the SpaceWIthGates interface

**Comment by valentinludu** [comment]
Killed it!
We did not need to load a space with gates anywhere.
The only place we need token gates list is the join page. Now, the search-domain and search-name responses are lighter and faster!

> I think there are other places where we get token gates, also please see if you can kill the SpaceWIthGates interface

File: charmClient/hooks/tokenGates.ts (hunk 1 above)
Comment (valentinludu) [reply charmClient/hooks/tokenGates.ts]: Done

**Review by mattcasey — APPROVED** [review]

**Review by mattcasey — COMMENTED** [review]
cleanup looks good

**Pull request merged**